src/main/java/de/jordanmruczynski/backend/service/CinemaService.java

```diff
@@ -0,0 +1,159 @@
+package de.jordanmruczynski.backend.service;
+
+import de.jordanmruczynski.backend.exception.FreePlaceBetweenSeatsException;
+import de.jordanmruczynski.backend.exception.ReservationTooLateException;
+import de.jordanmruczynski.backend.exception.ResourceNotFoundException;
+import de.jordanmruczynski.backend.exception.SeatAlreadyReservedException;
+import de.jordanmruczynski.backend.model.Reservation;
+import de.jordanmruczynski.backend.model.Screening;
+import de.jordanmruczynski.backend.model.Seat;
+import de.jordanmruczynski.backend.model.Ticket;
+import de.jordanmruczynski.backend.model.dto.*;
+import de.jordanmruczynski.backend.repository.*;
+import de.jordanmruczynski.backend.service.interfaces.ICinemaService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class CinemaService implements ICinemaService {
+
+    private final ScreeningRepository screeningRepository;
+    private final ReservationRepository reservationRepository;
+    private final SeatRepository seatRepository;
+    private final TicketRepository ticketRepository;
+
+    public CinemaService(ScreeningRepository screeningRepository, ReservationRepository reservationRepository, SeatRepository seatRepository, TicketRepository ticketRepository) {
+        this.screeningRepository = screeningRepository;
+        this.reservationRepository = reservationRepository;
+        this.seatRepository = seatRepository;
+        this.ticketRepository = ticketRepository;
+    }
+
+
+    @Override
+    public List<MovieScreeningsDTO> getAllScreenings(LocalDateTime start, LocalDateTime end) {
+        List<Screening> screenings = screeningRepository.findAllByScreeningStartTimeBetween(start, end);
+
+        screenings.sort(
```

Comment:
To sortowanie jest bardzo eleganckie, ale przez zmianę na mapę częściowo tracisz kolejność i musisz powtórzyć sortowanie niżej `.sorted(Map.Entry.comparingByKey())`

Reply:
```java
        final Map<String, List<LocalDateTime>> screeningsMap = screenings.stream()
                .collect(Collectors.groupingBy(
                        screening -> screening.getMovie().getTitle(),
                        TreeMap::new,
                        Collectors.mapping(Screening::getScreeningStartTime, Collectors.toList())
                ));

        return screeningsMap.entrySet().stream()
                .map(entry -> new MovieScreeningsDTO(entry.getKey(), entry.getValue()))
                .collect(Collectors.toList());
```
Czy coś takiego załatwi sprawe? (TreeMap)

Reply:
Bardzo sprytnie :)